### ISSUE TITLE
Interface prototype

### ISSUE THREAD

**Issue by mavimo** [Kindari/SublimeXdebug#8]
I create a simple mockup of interface to use XDebug.

A pop-up occur on variable user wont to inspect after some ms cursor is on variable AND is not moving. Popup display variable _var_dump_ result. Structured object ARE NOT navigable (all data are display) and data can't be changed (not change workflow).

![Mockup](http://i40.tinypic.com/vq5mqa.png)

Breakpoint are more visible using a red circle then an arrow, and I think is a good improvement highlight in a different color the current breakpoint (green?)

**Comment by Kindari** [comment]
The variable display won't be using var_dump. The DBGp protocol implements contexts, each context has properties, properties of certain types such as arrays and objects can have child properties. Contexts are also based on the current stack, which will be traversable.

Unless there is an unknown method to me, currently there is no way to do a popup with the sublime API. if I am wrong in that regard please let me know.

Current planned implementation is to create one view per context, contexts being Local, Superglobal, etc. One variable per line with its value and type. Double clicking on variables which contain child properties will open a new view, with those child properties displayed.

I like the proposed change for breakpoints and current line, but i'll go one step further and plan to make it configurable.

**Comment by Kindari** [comment]
Also, This thread's screenshot provides some influence on what I aim to do: http://www.sublimetext.com/forum/viewtopic.php?f=5&t=4819

**Comment by yriveiro** [comment]
+1

**Comment by franciscolourenco** [comment]
I agree with the red dots for breakpoins, but why don't you use an arrow for current line? Doesn't it make more sense?

**Comment by diaspar** [comment]
Hi. I tested and downloaded this plugin. I found it really easy to use, thanks. The major problem I found is that some objects and arrays are complex and the context view only shows 2 levels. Maybe sublime does not have a toooptip in their api, but maybe it has arrows and levels ( you can see them working on the file tree panel). That would be a good emulation of softwares like eclipse. If you can control how much you see of a complet var with arrows, it would be overkill.

My apologies I know little about sublime and python, otherwise I would send a pull request.

**Comment by Kindari** [comment]
@diaspar I would love to use the file tree for this system, but there is no api for it (at least to my knowledge).

One idea I've been toying with is using the Quick Panel system. It can be chained together infinitely, and is searchable.

In the meantime, you could vote here to get an api for the sidebar added! http://sublimetext.userecho.com/topic/92424-api-to-add-stuff-to-sidebar/
